src/aleph/vm/controllers/firecracker/executable.py

```diff
@@ -251,8 +260,37 @@ async def wait_for_init(self) -> None:
         May be empty."""
         return
 
+    def save_controller_configuration(self):
+        """Save VM configuration to be used by the controller service"""
+        with open(f"{settings.EXECUTION_ROOT}/{self.vm_hash}-controller.json", "wb") as controller_config_file:
+            controller_config_file.write(
+                self.controller_configuration.json(by_alias=True, exclude_none=True, indent=4).encode()
+            )
+            controller_config_file.flush()
+            config_file_path = Path(controller_config_file.name)
+            config_file_path.chmod(0o644)
+            return config_file_path
+
     async def configure(self):
-        raise NotImplementedError()
+        """Configure the VM by saving controller service configuration"""
+        if self.persistent:
+            firecracker_config_path = await self.fvm.save_configuration_file(self._firecracker_config)
+            vm_configuration = VMConfiguration(
+                firecracker_bin_path=self.fvm.firecracker_bin_path,
+                use_jailer=self.fvm.use_jailer,
+                jailer_bin_path=self.fvm.jailer_bin_path,
+                init_timeout=self.fvm.init_timeout,
+                config_file_path=firecracker_config_path,
+            )
+
+            configuration = Configuration(
+                vm_id=self.vm_id,
+                settings=settings,
+                vm_configuration=vm_configuration,
+            )
+
+            self.controller_configuration = configuration
+            self.save_controller_configuration()
```

Comment:
This pattern of saving the configuration in an attribute and then calling a function that uses that value just after seems weird, especially since that is the only place where that value is read.

Why not make it a separate function and call this ?

```suggestion
            save_controller_configuration(self.vm_hash, configuration)
```